### PR TITLE
fix parse resource in setting selector

### LINKED ISSUE #47718: kubectl set selector failed
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see https://kubernetes.io/docs/tasks/debug-application-cluster/troubleshooting/.):

**Note:** Please file issues for subcomponents under the appropriate repo

| Component | Repo                                                               |
| --------- | ------------------------------------------------------------------ |
| kubectl   | [kubernetes/kubectl](https://github.com/kubernetes/kubectl/issues/new) |
| kubeadm   | [kubernetes/kubeadm](https://github.com/kubernetes/kubeadm/issues/new) |

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Kubernetes version** (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"8+", GitVersion:"v1.8.0-alpha.0.823+a57c33bd281732-dirty", GitCommit:"a57c33bd2817323806b97da5c2f848108f7403d8", GitTreeState:"dirty", BuildDate:"2017-06-19T08:18:23Z", GoVersion:"go1.8.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"8+", GitVersion:"v1.8.0-alpha.0.823+a57c33bd281732-dirty", GitCommit:"a57c33bd2817323806b97da5c2f848108f7403d8", GitTreeState:"dirty", BuildDate:"2017-06-19T08:18:23Z", GoVersion:"go1.8.1", Compiler:"gc", Platform:"linux/amd64"}

**What happened**:
```
./cluster/kubectl.sh set selector service my-svc 'environment=qa'             
error: You must provide one or more resources by argument or filename.
Example resource specifications include:
   '-f rsrc.yaml'
   '--filename=rsrc.json'
   '<resource> <name>'
   '<resource>'
```

**What you expected to happen**:

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:

/sig cli

### PR DESCRIPTION
**Pull request by xilabao**
**What this PR does / why we need it**:

**Which issue this PR fixes**: fixes #47718

**Special notes for your reviewer**:

**Release note**:

```release-note
Fix "kubectl set selector" failing due to not being able to parse the resource type
```

### CODE REVIEW

**Comment by dims** [comment]
@k8s-bot ok to test

File: pkg/kubectl/cmd/set/set_selector.go [hunk 1]
```diff
@@ -116,6 +116,7 @@ func (o *SelectorOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args [
 	mapper, _ := f.Object()
 	o.mapper = mapper
 	o.encoder = f.JSONEncoder()
+	o.resources, o.selector, err = getResourcesAndSelector(args)
```
Comment (xiangpengzhao) [inline pkg/kubectl/cmd/set/set_selector.go]: Should it return if err != nil here?

Comment (xilabao) [inline pkg/kubectl/cmd/set/set_selector.go]: Done. thanks!

**Comment by deads2k** [comment]
@fabianofranz I think that @juanvallejo was in here recently, but I can't seem to choose him from the list

**Comment by juanvallejo** [comment]
@deads2k will add my review, thanks

File: hack/make-rules/test-cmd-util.sh [hunk 2]
```diff
@@ -2440,7 +2440,7 @@ run_service_tests() {
   # prove role=padawan
   kube::test::get_object_assert 'services redis-master' "{{range$service_selector_field}}{{.}}:{{end}}" "padawan:"
   # Set command to reset the selector back to the original one.
-  kubectl set selector -f examples/guestbook/redis-master-service.yaml app=redis,role=master,tier=backend
```
Comment (juanvallejo) [inline hack/make-rules/test-cmd-util.sh]: I think it's worth keeping this line, maybe also add a `--local` test?

**Comment by juanvallejo** [comment]
One comment, otherwise this lgtm

File: hack/make-rules/test-cmd-util.sh (hunk 2 above)
Comment (xilabao) [inline hack/make-rules/test-cmd-util.sh]: Done. PTAL @juanvallejo

**Review by juanvallejo — APPROVED** [review]

**Pull request merged**